tests/framework/unit_tests/Optimizers/testSwapMutator.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+  Testing for the SwapMutator method
+  @authors: Junyung Kim and Mohammad Abdo
+"""
+
+
+import os
+import sys
+import xarray as xr
+import numpy as np
+
+ravenPath = os.path.abspath(os.path.join(__file__, *['..'] * 5))
+print('... located RAVEN at:', ravenPath)
+sys.path.append(ravenPath)
+from ravenframework.CustomDrivers import DriverUtils
+DriverUtils.doSetup()
+
+from ravenframework.Optimizers.mutators.mutators import returnInstance
+import xml.etree.ElementTree as ET
+from ravenframework import MessageHandler
+
+from ravenframework import Distributions
+from ravenframework.Distributions import UniformDiscrete
+from ravenframework.Distributions import Categorical
+
+mh = MessageHandler.MessageHandler()
+mh.initialize({'verbosity':'debug'})
+
+def checkSameDataArrays(comment, resultedDA, expectedDA, update=True):
+  """
+    This method compares two identical things
+    @ In, comment, string, a comment printed out if it fails
+    @ In, resultedDA, xr.DataArray, the resulted DataArray to be tested
+    @ In, expectedDA, xr.DataArray, the expected DataArray
+    @ In, update, bool, optional, if False then don't update results counter
+    @ Out, res, bool, True if same
+  """
+  res = resultedDA.identical(expectedDA)
+  if update:
+    if res:
+      results["pass"] += 1
+    else:
+      print("checking string", comment, '|', resultedDA, "!=", expectedDA)
+      results["fail"] += 1
+  return res
+
+results = {'pass': 0, 'fail': 0}
+
+def createElement(tag,attrib={},text={}):
+  """
+    Method to create a dummy xml element readable by the distribution classes
+    @ In, tag, string, the node tag
+    @ In, attrib, dict, optional, the attribute of the xml node
+    @ In, text, dict, optional, the dict containing what should be in the xml text
+  """
+  element = ET.Element(tag,attrib)
+  element.text = text
+  return element
```

Comment:
Not related to this PR. I think we should start to standardize our unit tests. The first thing to do is to create utils functions for unit tests. For examples, the above functions have been existed in a lot of unit test files.